pgjdbc/src/test/java/org/postgresql/test/util/PasswordUtilTest.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2023, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.test.util;
+
+import org.postgresql.PGConnection;
+import org.postgresql.core.Utils;
+import org.postgresql.test.TestUtil;
+import org.postgresql.test.util.rules.annotation.HaveMinimalServerVersion;
+import org.postgresql.util.PasswordUtil;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.security.SecureRandom;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.Properties;
+
+public class PasswordUtilTest {
+  private static final SecureRandom rng = new SecureRandom();
+
+  private static String randomSuffix() {
+    return "" + Math.abs(rng.nextLong());
```

Comment:
It can still be negative in case of Long.MIN_VALUE.
Something like Long.toUnsignedString or toHexString might be better here

Reply:
Ha!  I had this using the newer `rng.nextLong(0, Long.MAX_VALUE)` but it's not in the older JDKs. When I changed it to `Math.abs(...)`, should I really care about that 1/2^63 chance in the suite?

Yes, `Long.toHexString(...)` is a fine choice here.